packages/editor/src/components/autosave-monitor/test/index.js

```diff
@@ -82,6 +82,28 @@ describe( 'AutosaveMonitor', () => {
 
 			expect( toggleTimer ).toHaveBeenCalledWith( false );
 		} );
+
+		it( 'should avoid scheduling autosave if still dirty but already autosaved for edits', () => {
+			// Explanation: When a published post is autosaved, it's still in a
+			// dirty state since the edits are not saved to the post until the
+			// user clicks "Update". To avoid recurring autosaves, ensure that
+			// an edit has occurred since the last autosave had completed.
+
+			const beforeReference = [];
+			const afterReference = [];
+
+			// A post is non-dirty while autosave is in-flight.
+			wrapper.setProps( { isDirty: false, isAutosaving: true, isAutosaveable: true, editsReference: beforeReference } );
+			toggleTimer.mockClear();
+			wrapper.setProps( { isDirty: true, isAutosaving: false, isAutosaveable: true, editsReference: beforeReference } );
+
+			expect( toggleTimer ).toHaveBeenCalledWith( false );
+
+			// Once edit occurs after autosave, resume scheduling.
+			wrapper.setProps( { isDirty: true, isAutosaving: false, isAutosaveable: true, editsReference: afterReference } );
+
+			expect( toggleTimer ).toHaveBeenCalledWith( false );
```

Comment:
Maybe I'm mis-reading the test, but I would expect this to test for something different. Now the post is dirty, isn't autosaving, is autosavable, and has a different `editsReference`, but the expectation is the same.

I don't quite follow 🤷‍♂️

Reply:
> Maybe I'm mis-reading the test, but I would expect this to test for something different.

You're totally right. We expect `toggleTimer` to have been called with `true`, which it _is_, but as written `toHaveBeenCalledWith` would consider both this and the prior invocation (since we didn't clear mock calls between). So while it's true that it had been called with `false`, this was not what we're testing.

I've updated the test so that it explicitly checks that the second time around, it's called with `true`.